aggregator/src/aggregator.rs

```diff
@@ -2524,95 +2543,107 @@ impl VdafOps {
                         ));
                     }
 
-                    if let Some(collect_job_id) = tx
-                        .get_collect_job_id::<L, Q, A>(
-                            task.id(),
-                            &batch_identifier,
-                            &aggregation_param,
-                        )
-                        .await?
+                    // Check if this collection job already exists, ensuring that all parameters match.
+                    if let Some(collection_job) =
+                        tx.get_collection_job::<L, Q, A>(&collection_id).await?
                     {
-                        debug!(collect_request = ?req, "Serving existing collect job UUID");
-                        return Ok(collect_job_id);
+                        if collection_job.batch_identifier() == &batch_identifier
```

Comment:
If the checks are different enough (and it sounds like they are), this might be a good place to introduce another method on `CollectableQueryType` which performs this validation.

(also, I believe this isn't caused by this PR; maybe it's better to write up an issue & address in a later PR since this PR is already huge?)